src/common/selectors/pipelineStageSelector.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Connection } from '@salesforce/core';
+import { QueryResult } from 'jsforce';
+import { PipelineStage } from '../types';
+
+/**
+ *
+ * Returns an array of a pipeline stages filtering by project name.
+ */
+export async function selectPipelineStagesByProject(con: Connection, projectName: string): Promise<PipelineStage[]> {
+  const queryStr = `SELECT Id, sf_devops__Pipeline__r.sf_devops__Project__c, sf_devops__Branch__r.sf_devops__Name__c, (SELECT Id FROM sf_devops__Pipeline_Stages__r)
```

Comment:
We could also filter this query by branch name to get only the correct pipeline stage. But if the result is empty we can't know which one was wrong, the project name or the branch name. And it was a requirement to be very clear in the error message.

Reply:
Can we validate the stage we found, to make sure it has the correct branch name?